File: database-driver-plugin/src/main/java/de/lino/database/database/sql/orcale/OracleSQLDatabaseProvider.java
package de.lino.database.database.sql.orcale;

import de.lino.database.database.auth.Credentials;
import de.lino.database.database.DatabaseType;
import de.lino.database.database.sql.SQLDatabaseProvider;
import de.lino.database.database.sql.SQLExecution;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * The {@link SQLDatabaseProvider} for Oracle Database, connected via {@link SQLExecution} using
 * {@link DatabaseType#ORACLE}'s JDBC driver and URL scheme.
 */
@Getter
public class OracleSQLDatabaseProvider extends SQLDatabaseProvider {

    /**
     * Connects to an Oracle database with {@code credentials}.
     *
     * @param credentials the login credentials and connection details to connect with
     */
    public OracleSQLDatabaseProvider(@NotNull Credentials credentials) {
        super(DatabaseType.ORACLE, new SQLExecution(DatabaseType.ORACLE, credentials));
    }
    
}

File: python/database-driver-plugin/src/database_driver/plugin/database/sql/orcale/oracle_sql_database_provider.py
"""Mirror of ``de.lino.database.database.sql.orcale.OracleSQLDatabaseProvider``."""

from __future__ import annotations

from database_driver.api.database.auth.credentials import Credentials
from database_driver.api.database.database_type import DatabaseType
from database_driver.plugin.database.sql.sql_database_provider import SQLDatabaseProvider
from database_driver.plugin.database.sql.sql_execution import SQLExecution


class OracleSQLDatabaseProvider(SQLDatabaseProvider):
    """The ``SQLDatabaseProvider`` for Oracle Database (thin mode), connected via ``SQLExecution`` using
    ``DatabaseType.ORACLE``'s driver and connection scheme."""

    def __init__(self, credentials: Credentials) -> None:
        super().__init__(DatabaseType.ORACLE, SQLExecution(DatabaseType.ORACLE, credentials))
